build, qt: Make Qt rcc output always deterministic

hebasto opened the pull request:
The Qt Resource Compiler ([rcc](https://doc.qt.io/qt-5/rcc.html)) has a command-line option `--format-version` which has the [default value](https://code.qt.io/cgit/qt/qtbase.git/tree/src/tools/rcc/main.cpp?h=5.12.10#n172) 2.

The only difference from `--format-version 1` is adding a [last modified timestamp](https://code.qt.io/cgit/qt/qtbase.git/tree/src/tools/rcc/rcc.cpp?h=5.12.10#n207) to the output file ([credits](https://github.com/bitcoin/bitcoin/pull/21654#issuecomment-819198228) to **fanquake**). That, in turn, forces us to use `QT_RCC_SOURCE_DATE_OVERRIDE=1` to get deterministic builds (#13732).

This change makes rcc output always deterministic by using `--format-version 1` option that makes usage of the
`QT_RCC_SOURCE_DATE_OVERRIDE` needless.

---

Also it improves interaction with ccache:

On master (f6c44e999b7d1d9a0de5d678ac8f1679aa271f65):
```
$ make && make clean && ccache --zero-stats && make && ccache --show-stats
...
cache directory                     /home/hebasto/.ccache
primary config                      /home/hebasto/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
stats updated                       Sun Apr 11 15:45:43 2021
stats zeroed                        Sun Apr 11 15:45:05 2021
cache hit (direct)                   638
cache hit (preprocessed)               0
cache miss                             1
cache hit rate                     99.84 %
called for link                       10
cleanups performed                     0
files in cache                     20023
cache size                          13.2 GB
max cache size                      15.0 GB
```

The missed file is always `qt/libbitcoinqt_a-qrc_bitcoin_locale.o`.

With this PR:
```
$ make && make clean && ccache --zero-stats && make && ccache --show-stats
...
cache directory                     /home/hebasto/.ccache
primary config                      /home/hebasto/.ccache/ccache.conf
secondary config      (readonly)    /etc/ccache.conf
stats updated                       Sun Apr 11 15:28:46 2021
stats zeroed                        Sun Apr 11 15:28:21 2021
cache hit (direct)                   639
cache hit (preprocessed)               0
cache miss                             0
cache hit rate                    100.00 %
called for link                       10
cleanups performed                     0
files in cache                     20012
cache size                          13.2 GB
max cache size                      15.0 GB
```

hebasto committed:
build: Makes rcc output always deterministic

The Qt Resource Compiler (rcc) has a command-line option
`--format-version` which has the default value 2.

The only difference from `--format-version 1` is adding a last modified
timestamp to the output file. That, in turn, forces us to use
`QT_RCC_SOURCE_DATE_OVERRIDE=1` to get deterministic builds.

This change makes rcc output always deterministic by using
`--format-version 1` option that makes usage of the
`QT_RCC_SOURCE_DATE_OVERRIDE` needless. Also it improves interaction
with ccache.

Co-authored-by: fanquake <fanquake@gmail.com>

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
5d0c5016f27b1e3122a38dbe32ece19543a7fb01c94a6da5146f017cea81efa2  guix-build-bbd54508e186/output/aarch64-linux-gnu/bitcoin-bbd54508e186-aarch64-linux-gnu-debug.tar.gz
1694738123d5bd4af625f38eaa17b7ba1df223aa4e3471f53a5d081d3a8cc7ed  guix-build-bbd54508e186/output/aarch64-linux-gnu/bitcoin-bbd54508e186-aarch64-linux-gnu.tar.gz
d62e913e95a6843b6cb84d607dca6db832cb55a3e286a9a739dd1b17d58a8bb1  guix-build-bbd54508e186/output/arm-linux-gnueabihf/bitcoin-bbd54508e186-arm-linux-gnueabihf-debug.tar.gz
37af054550f202f307f6c031f88f3ef23b39bbba4a4b6cda850f0a04048ceb8e  guix-build-bbd54508e186/output/arm-linux-gnueabihf/bitcoin-bbd54508e186-arm-linux-gnueabihf.tar.gz
9f7fec2708b006635646ca5f259abbd96a99cb5a2b2b6ee7fbc15de07499d2ee  guix-build-bbd54508e186/output/dist-archive/bitcoin-bbd54508e186.tar.gz
de142857b2ea16b4a6b6c9cb67c169dc79020e07bf90fc4dd48c5eceba34cbce  guix-build-bbd54508e186/output/powerpc64-linux-gnu/bitcoin-bbd54508e186-powerpc64-linux-gnu-debug.tar.gz
3f3e6e1dac5d35cd70310fd966edd257bce4d6e58a7c0f5481bf41bfe2a32d09  guix-build-bbd54508e186/output/powerpc64-linux-gnu/bitcoin-bbd54508e186-powerpc64-linux-gnu.tar.gz
c7111777890122068de13006c13048a7fad2c3761f367949639b30401d2573fe  guix-build-bbd54508e186/output/powerpc64le-linux-gnu/bitcoin-bbd54508e186-powerpc64le-linux-gnu-debug.tar.gz
967bf59cc18478b50c6563d32b1f34bac4d22e9b21f3af580d0704cdf1479ae7  guix-build-bbd54508e186/output/powerpc64le-linux-gnu/bitcoin-bbd54508e186-powerpc64le-linux-gnu.tar.gz
4126161aadf7c2eef8070e2a8a9c6064f9d66fe220261db4cf2f6c668e0f1ec2  guix-build-bbd54508e186/output/riscv64-linux-gnu/bitcoin-bbd54508e186-riscv64-linux-gnu-debug.tar.gz
b9a17f4bb45c5d19ea2085da2703057bc4cbdf9297004712382f593aa6150201  guix-build-bbd54508e186/output/riscv64-linux-gnu/bitcoin-bbd54508e186-riscv64-linux-gnu.tar.gz
c589a046701b6eb27fef10ebb5f1c637f3419483cfffdc2cb0a30396777f03c5  guix-build-bbd54508e186/output/x86_64-apple-darwin18/bitcoin-bbd54508e186-osx-unsigned.dmg
dd1cf0e945fc47a59d760688aa11db877f5267a3bd5df45393edc3992b5a2912  guix-build-bbd54508e186/output/x86_64-apple-darwin18/bitcoin-bbd54508e186-osx-unsigned.tar.gz
a799066051cc52bcee54bd8e4f513a97c4fad3ecd237c20711b50e8b02384e95  guix-build-bbd54508e186/output/x86_64-apple-darwin18/bitcoin-bbd54508e186-osx64.tar.gz
446ce5a65a7671d61d5f7a81dd908413059531967ec6245c5847fc99cedcdca9  guix-build-bbd54508e186/output/x86_64-linux-gnu/bitcoin-bbd54508e186-x86_64-linux-gnu-debug.tar.gz
f7c8c35a4cbd0fbf8e4ed2cef34c31d817f17dfda5114ec9536d1aee4cc9f774  guix-build-bbd54508e186/output/x86_64-linux-gnu/bitcoin-bbd54508e186-x86_64-linux-gnu.tar.gz
35be25ad31ebf3dc90b1f513a269974b650ba242e4dd2ac5ea41de549988513e  guix-build-bbd54508e186/output/x86_64-w64-mingw32/bitcoin-bbd54508e186-win-unsigned.tar.gz
47a0e86ed18af195c051e2e67f4f5b55d6ef8d0d6fc4240ef95483975414342b  guix-build-bbd54508e186/output/x86_64-w64-mingw32/bitcoin-bbd54508e186-win64-debug.zip
ccf64b3eeac2725da079121da4df07830e94002af1e56dcc8a68c91ce94b2992  guix-build-bbd54508e186/output/x86_64-w64-mingw32/bitcoin-bbd54508e186-win64-setup-unsigned.exe
c752ae831b88c758989ae8c93ffadfc06572520105ebb364a33e1d3d8e19b206  guix-build-bbd54508e186/output/x86_64-w64-mingw32/bitcoin-bbd54508e186-win64.zip
```

hebasto commented:
#### Gitian builds:
- Linux
```
Generating report
e48be20bc7cca4ce97f92385c07d83a2da8647690823f046419cc1ec451a1b54  bitcoin-bbd54508e186-aarch64-linux-gnu-debug.tar.gz
af095b9f3be2d84087604e255df3753715eaaf3d72f0e231db29e0f9f071536f  bitcoin-bbd54508e186-aarch64-linux-gnu.tar.gz
fe2aa3033a5cc59914b1ac091fefe8fef248d05f46b032888cb7806bc9327413  bitcoin-bbd54508e186-arm-linux-gnueabihf-debug.tar.gz
61441c05c0f48525d9dfcf045233ac69eed186e3915f028fa747f2cd09866ce3  bitcoin-bbd54508e186-arm-linux-gnueabihf.tar.gz
d0b39207eafb279d70b0dd519990896161d687075c281ef3934c677e70cb907c  bitcoin-bbd54508e186-powerpc64-linux-gnu-debug.tar.gz
8b2f4644d42368e9af6a2b6b161d3ef112c08dda048fc573163de58acb492038  bitcoin-bbd54508e186-powerpc64-linux-gnu.tar.gz
2d99869e5585a7593414f479281cb3bc7f5953f829343aecfb7654caf9cafea5  bitcoin-bbd54508e186-powerpc64le-linux-gnu-debug.tar.gz
cd0f629eaa113d3d8e24c898345018f33d562653239a63ef930b280c61ce2eae  bitcoin-bbd54508e186-powerpc64le-linux-gnu.tar.gz
236cd9b9f41b1e400c8c32c81b3cdb8eabaac5583990128edfa6cf0d95e89ff5  bitcoin-bbd54508e186-riscv64-linux-gnu-debug.tar.gz
3003f8a4f372f784402e6f79354138138d7adfa858d018910164ef053e6acdae  bitcoin-bbd54508e186-riscv64-linux-gnu.tar.gz
2e38ed986aa634745a8382a8d546bb5fb23e496b26a7b6aef9205d73c5c17365  bitcoin-bbd54508e186-x86_64-linux-gnu-debug.tar.gz
df5cf91785c587fb0fc100299b160a97415f36ae93153f2a9b93e140d84a3de5  bitcoin-bbd54508e186-x86_64-linux-gnu.tar.gz
9f7fec2708b006635646ca5f259abbd96a99cb5a2b2b6ee7fbc15de07499d2ee  src/bitcoin-bbd54508e186.tar.gz
40ec5fa387326baf0634876347d776b42586ff5effa1dfd407c11fadb79f636f  bitcoin-core-linux-22-res.yml
Done.
```
- Windows
```
Generating report
d4fa0dd29ddae91af607ae5b0932d427f141dfbc961d587b45854b1b54ecd208  bitcoin-bbd54508e186-win-unsigned.tar.gz
3731b1d39196378c2dcb0c118f5946263305a8bf97eed0b1e1af44a0b2c89bbd  bitcoin-bbd54508e186-win64-debug.zip
efcbad96bc3975f01aa4f7806828cbc9385cb0e94ee845b42760f2f260ee8ec8  bitcoin-bbd54508e186-win64-setup-unsigned.exe
aa8aa33662ef7cf2e54cee8bd8019793ee223b416afa021567549ffe1fa4110b  bitcoin-bbd54508e186-win64.zip
9f7fec2708b006635646ca5f259abbd96a99cb5a2b2b6ee7fbc15de07499d2ee  src/bitcoin-bbd54508e186.tar.gz
1ec32216f323f47e7475a2211df145d85805782ff59d1c89fe2a1de7ead82088  bitcoin-core-win-22-res.yml
Done.
```
- macOS
```
Generating report
9a5a01c917c43c3f7c8ada3c071c4112b54b28a2c295b0c098a241b6d3c8d5fd  bitcoin-bbd54508e186-osx-unsigned.dmg
5a5f37ea9a3c1f469e245bcd20a55cca75f9a1c1b16d6675a33075da61247748  bitcoin-bbd54508e186-osx-unsigned.tar.gz
74565dc89b87f134288f975e1f48099a53c892b7b25595a9299d333a5219e10c  bitcoin-bbd54508e186-osx64.tar.gz
9f7fec2708b006635646ca5f259abbd96a99cb5a2b2b6ee7fbc15de07499d2ee  src/bitcoin-bbd54508e186.tar.gz
a72e462fe8a1a3a6d1bdf4a469e72ba091b30ef3d8a928c08e8dcd2c48f93be5  bitcoin-core-osx-22-res.yml
Done.
```

jarolrod reviewed (commented):
Contributing GUIX hashes, mine match @hebasto 

```
find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

5d0c5016f27b1e3122a38dbe32ece19543a7fb01c94a6da5146f017cea81efa2  guix-build-bbd54508e186/output/aarch64-linux-gnu/bitcoin-bbd54508e186-aarch64-linux-gnu-debug.tar.gz
1694738123d5bd4af625f38eaa17b7ba1df223aa4e3471f53a5d081d3a8cc7ed  guix-build-bbd54508e186/output/aarch64-linux-gnu/bitcoin-bbd54508e186-aarch64-linux-gnu.tar.gz
d62e913e95a6843b6cb84d607dca6db832cb55a3e286a9a739dd1b17d58a8bb1  guix-build-bbd54508e186/output/arm-linux-gnueabihf/bitcoin-bbd54508e186-arm-linux-gnueabihf-debug.tar.gz
37af054550f202f307f6c031f88f3ef23b39bbba4a4b6cda850f0a04048ceb8e  guix-build-bbd54508e186/output/arm-linux-gnueabihf/bitcoin-bbd54508e186-arm-linux-gnueabihf.tar.gz
9f7fec2708b006635646ca5f259abbd96a99cb5a2b2b6ee7fbc15de07499d2ee  guix-build-bbd54508e186/output/dist-archive/bitcoin-bbd54508e186.tar.gz
de142857b2ea16b4a6b6c9cb67c169dc79020e07bf90fc4dd48c5eceba34cbce  guix-build-bbd54508e186/output/powerpc64-linux-gnu/bitcoin-bbd54508e186-powerpc64-linux-gnu-debug.tar.gz
3f3e6e1dac5d35cd70310fd966edd257bce4d6e58a7c0f5481bf41bfe2a32d09  guix-build-bbd54508e186/output/powerpc64-linux-gnu/bitcoin-bbd54508e186-powerpc64-linux-gnu.tar.gz
c7111777890122068de13006c13048a7fad2c3761f367949639b30401d2573fe  guix-build-bbd54508e186/output/powerpc64le-linux-gnu/bitcoin-bbd54508e186-powerpc64le-linux-gnu-debug.tar.gz
967bf59cc18478b50c6563d32b1f34bac4d22e9b21f3af580d0704cdf1479ae7  guix-build-bbd54508e186/output/powerpc64le-linux-gnu/bitcoin-bbd54508e186-powerpc64le-linux-gnu.tar.gz
4126161aadf7c2eef8070e2a8a9c6064f9d66fe220261db4cf2f6c668e0f1ec2  guix-build-bbd54508e186/output/riscv64-linux-gnu/bitcoin-bbd54508e186-riscv64-linux-gnu-debug.tar.gz
b9a17f4bb45c5d19ea2085da2703057bc4cbdf9297004712382f593aa6150201  guix-build-bbd54508e186/output/riscv64-linux-gnu/bitcoin-bbd54508e186-riscv64-linux-gnu.tar.gz
c589a046701b6eb27fef10ebb5f1c637f3419483cfffdc2cb0a30396777f03c5  guix-build-bbd54508e186/output/x86_64-apple-darwin18/bitcoin-bbd54508e186-osx-unsigned.dmg
dd1cf0e945fc47a59d760688aa11db877f5267a3bd5df45393edc3992b5a2912  guix-build-bbd54508e186/output/x86_64-apple-darwin18/bitcoin-bbd54508e186-osx-unsigned.tar.gz
a799066051cc52bcee54bd8e4f513a97c4fad3ecd237c20711b50e8b02384e95  guix-build-bbd54508e186/output/x86_64-apple-darwin18/bitcoin-bbd54508e186-osx64.tar.gz
446ce5a65a7671d61d5f7a81dd908413059531967ec6245c5847fc99cedcdca9  guix-build-bbd54508e186/output/x86_64-linux-gnu/bitcoin-bbd54508e186-x86_64-linux-gnu-debug.tar.gz
f7c8c35a4cbd0fbf8e4ed2cef34c31d817f17dfda5114ec9536d1aee4cc9f774  guix-build-bbd54508e186/output/x86_64-linux-gnu/bitcoin-bbd54508e186-x86_64-linux-gnu.tar.gz
35be25ad31ebf3dc90b1f513a269974b650ba242e4dd2ac5ea41de549988513e  guix-build-bbd54508e186/output/x86_64-w64-mingw32/bitcoin-bbd54508e186-win-unsigned.tar.gz
47a0e86ed18af195c051e2e67f4f5b55d6ef8d0d6fc4240ef95483975414342b  guix-build-bbd54508e186/output/x86_64-w64-mingw32/bitcoin-bbd54508e186-win64-debug.zip
ccf64b3eeac2725da079121da4df07830e94002af1e56dcc8a68c91ce94b2992  guix-build-bbd54508e186/output/x86_64-w64-mingw32/bitcoin-bbd54508e186-win64-setup-unsigned.exe
c752ae831b88c758989ae8c93ffadfc06572520105ebb364a33e1d3d8e19b206  guix-build-bbd54508e186/output/x86_64-w64-mingw32/bitcoin-bbd54508e186-win64.zip
```

fanquake commented:
Does this still need setting in the qt build env if it's now being done in the makefile? If so you should probably note why that's the case:

https://github.com/bitcoin/bitcoin/blob/e7af2f35af95f4ca51e38c8ac5b05cad8be22489/depends/packages/qt.mk#L177

I assume we could also skip setting `QT_RCC_SOURCE_DATE_OVERRIDE` entirely by using the version 1 format of the generated files. i.e pass `--format-version 1` to `rcc`. From my read of the Qt source, the only thing added in version 2 is the last modified time, which we obviously don't care about.

hebasto commented:
@fanquake 
> Does this still need setting in the qt build env if it's now being done in the makefile?

No. The determinism is preserved.

> I assume we could also skip setting `QT_RCC_SOURCE_DATE_OVERRIDE` entirely by using the version 1 format of the generated files. i.e pass `--format-version 1` to `rcc`. From my read of the Qt source, the only thing added in version 2 is the last modified time, which we obviously don't care about.

Thanks!. I checked the Qt source code, and implemented your suggestion.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
2c4f3cfd5ca8730506c5b1bc58fa5e6d447c889f2924ed74b7f0044b3b1486eb  guix-build-6c05fa9bfcad/output/aarch64-linux-gnu/bitcoin-6c05fa9bfcad-aarch64-linux-gnu-debug.tar.gz
9f6cfe70010fc4aeba8d58508b3b6994916497882da0375c34a751f421d93efe  guix-build-6c05fa9bfcad/output/aarch64-linux-gnu/bitcoin-6c05fa9bfcad-aarch64-linux-gnu.tar.gz
b6100bc3d08e2ef992c43e58eeefc0116d2dd6e3bb57e61cb693c5f320d98b1b  guix-build-6c05fa9bfcad/output/arm-linux-gnueabihf/bitcoin-6c05fa9bfcad-arm-linux-gnueabihf-debug.tar.gz
f8a794defe4e49c2e5d0fd4329cbf1ed34060f07b65be0aefabc0663deef5e21  guix-build-6c05fa9bfcad/output/arm-linux-gnueabihf/bitcoin-6c05fa9bfcad-arm-linux-gnueabihf.tar.gz
29228571cd7826692a1a87c42705b5f9cf0d5cc5b39e659fd0ecba8647323dc7  guix-build-6c05fa9bfcad/output/dist-archive/bitcoin-6c05fa9bfcad.tar.gz
031f28b6976f736b3949659523ab82e6540b647846d88ed7354cf7219dd54fab  guix-build-6c05fa9bfcad/output/powerpc64-linux-gnu/bitcoin-6c05fa9bfcad-powerpc64-linux-gnu-debug.tar.gz
44a1a0b0dc7d41590b84e39f7629c50a10bc0264172fa9e15b33a3e9f7be9709  guix-build-6c05fa9bfcad/output/powerpc64-linux-gnu/bitcoin-6c05fa9bfcad-powerpc64-linux-gnu.tar.gz
aa4793840f362382dbc6f9eadf987d603dc23d36c8b776be849f3efd06690d4c  guix-build-6c05fa9bfcad/output/powerpc64le-linux-gnu/bitcoin-6c05fa9bfcad-powerpc64le-linux-gnu-debug.tar.gz
35f9b831f1ec7d96dcd8d86c94a88ab3e29d9dd2d0ead74135e8cf4073c94ae4  guix-build-6c05fa9bfcad/output/powerpc64le-linux-gnu/bitcoin-6c05fa9bfcad-powerpc64le-linux-gnu.tar.gz
9cbcbc1561ee0dbd849a6fc4d06f5bb598a8710beee82d8e6c5c16b695c2e5ed  guix-build-6c05fa9bfcad/output/riscv64-linux-gnu/bitcoin-6c05fa9bfcad-riscv64-linux-gnu-debug.tar.gz
6eff3d8af9b3c42925579cf8611129d30422c9a6f8f23a1f6f7b2c26b611bfba  guix-build-6c05fa9bfcad/output/riscv64-linux-gnu/bitcoin-6c05fa9bfcad-riscv64-linux-gnu.tar.gz
1dd4c170b0bd56bf129207db94ad5e09caec1baba829e44789236a4e93450c33  guix-build-6c05fa9bfcad/output/x86_64-apple-darwin18/bitcoin-6c05fa9bfcad-osx-unsigned.dmg
316591d72893ed608cd93d6a524cc2e2c86b25f6a08a3fb08ae4a66ee97a9701  guix-build-6c05fa9bfcad/output/x86_64-apple-darwin18/bitcoin-6c05fa9bfcad-osx-unsigned.tar.gz
8eefa59f672b7cbcb827308dcf0b04217304479228fcc629c54aee90bb52329c  guix-build-6c05fa9bfcad/output/x86_64-apple-darwin18/bitcoin-6c05fa9bfcad-osx64.tar.gz
47ecc757ff152fca1c976783eba32979d902d1babf9502a58dc52ebcfc5c20f5  guix-build-6c05fa9bfcad/output/x86_64-linux-gnu/bitcoin-6c05fa9bfcad-x86_64-linux-gnu-debug.tar.gz
cbe56fdef102f6c42e27cccd45126e7e58718c6a3cc79cee8abc400697f2f79b  guix-build-6c05fa9bfcad/output/x86_64-linux-gnu/bitcoin-6c05fa9bfcad-x86_64-linux-gnu.tar.gz
1ced659f670c8bc599b79a188db522d9772ac2627a3018f4f859b73a8c85085a  guix-build-6c05fa9bfcad/output/x86_64-w64-mingw32/bitcoin-6c05fa9bfcad-win-unsigned.tar.gz
59656bc928a67ecc1c4e165cf235fa2c0db79517362b10b04c8482effcdf5d1a  guix-build-6c05fa9bfcad/output/x86_64-w64-mingw32/bitcoin-6c05fa9bfcad-win64-debug.zip
b865fb5d01679658f4f2d3a3310b9131b6f107cd2900e1420853aeca4b84750f  guix-build-6c05fa9bfcad/output/x86_64-w64-mingw32/bitcoin-6c05fa9bfcad-win64-setup-unsigned.exe
7f341c7ce4e7d3710f273f5b6bda71b901136819f32197b9b29d178ce981e083  guix-build-6c05fa9bfcad/output/x86_64-w64-mingw32/bitcoin-6c05fa9bfcad-win64.zip
```
#### Gitian builds:
- Linux
```
Generating report
5ca160e3a8c80eb616a59658340b976046e8374fb3ec942f741d7b364704bc20  bitcoin-6c05fa9bfcad-aarch64-linux-gnu-debug.tar.gz
64a8c10f2729d56e94114ff76468f3de98867b0d104ed3354ead753510560faf  bitcoin-6c05fa9bfcad-aarch64-linux-gnu.tar.gz
b09161f9a1d8a9d7ce6697220cfd08257b16c350282ae84e76e475c214babb43  bitcoin-6c05fa9bfcad-arm-linux-gnueabihf-debug.tar.gz
546f70493349809ca7ac7e1cb647ed1274f4939e6e1df48d0fcda45129244ede  bitcoin-6c05fa9bfcad-arm-linux-gnueabihf.tar.gz
4419a88bd74a5fe69abce62e590914e705fc58d64097fcb16db25dc177006b6e  bitcoin-6c05fa9bfcad-powerpc64-linux-gnu-debug.tar.gz
24890070fda8ede09898bcca47157ae1cb6d92f6d987e29a9dcf4cdac46a86b6  bitcoin-6c05fa9bfcad-powerpc64-linux-gnu.tar.gz
a83ec6d43d3c7b3748343483fd22063e0238dc77d9e3d188ec4f25a57fdf7d98  bitcoin-6c05fa9bfcad-powerpc64le-linux-gnu-debug.tar.gz
7f4b3bfb3094645f4a96e73a13ccaa0818b9cd76b2bb087b28f81065b31eeaff  bitcoin-6c05fa9bfcad-powerpc64le-linux-gnu.tar.gz
8dc8b0048fbe37ee3bb1a6628fb3de29f0b9f45e017debb9c2f8c6f95fd60db0  bitcoin-6c05fa9bfcad-riscv64-linux-gnu-debug.tar.gz
2de0db68d0abab07bd5acf844cd1c9ebf37a84eb40c2c1f20f2b78f34ce09936  bitcoin-6c05fa9bfcad-riscv64-linux-gnu.tar.gz
8db61f4a126f59f33e205d4b7008c73a78103081abfbf794a985ef4ee7ecb399  bitcoin-6c05fa9bfcad-x86_64-linux-gnu-debug.tar.gz
d260df861ec65a7145258934efb69f5225afe0fff03393aaedf5702d50ef7a45  bitcoin-6c05fa9bfcad-x86_64-linux-gnu.tar.gz
29228571cd7826692a1a87c42705b5f9cf0d5cc5b39e659fd0ecba8647323dc7  src/bitcoin-6c05fa9bfcad.tar.gz
dec87960d3bc9ca50793a083d9de94dbe11405c1b2e20b368d1cc8cc0073bb7c  bitcoin-core-linux-22-res.yml
Done.
```
- Windows
```
Generating report
9afd1c5a1a0b906fa48e8ba744caaea22ece1a76aa51a9fcf289d82a8095adab  bitcoin-6c05fa9bfcad-win-unsigned.tar.gz
015c5fbf9f7af8309c6b96979a6aaf85eb0a1e9412bbde1ac73f507ffb1283db  bitcoin-6c05fa9bfcad-win64-debug.zip
01a226d2f9128bb7c5ed99059f4a40e0028f4a538ddb016dc2360951799b4bf1  bitcoin-6c05fa9bfcad-win64-setup-unsigned.exe
8fd2c8b81f65d0644fa1d7173b7ee7d5871608cc589c95fe6084702709800e48  bitcoin-6c05fa9bfcad-win64.zip
29228571cd7826692a1a87c42705b5f9cf0d5cc5b39e659fd0ecba8647323dc7  src/bitcoin-6c05fa9bfcad.tar.gz
8bc9a5f3c4dc7e51e2621cbaa3098a99451093a9bb8b5429b928549099cfb89e  bitcoin-core-win-22-res.yml
Done.
```
- macOS
```
Generating report
24be7df3f99bfb9b730e8539c9165e7a42d7bfd929581dcd8ea3651c2e30a0da  bitcoin-6c05fa9bfcad-osx-unsigned.dmg
4778461cf8658c53c7b9815e7faa5158f2b013260b31e3bff77de58d9e179eac  bitcoin-6c05fa9bfcad-osx-unsigned.tar.gz
c2594718b499b45ca7f4a7e98f10682391948cb5e8e3a9afd1b7347e5567c7d1  bitcoin-6c05fa9bfcad-osx64.tar.gz
29228571cd7826692a1a87c42705b5f9cf0d5cc5b39e659fd0ecba8647323dc7  src/bitcoin-6c05fa9bfcad.tar.gz
d08a3ad67e829a7b34363c0963766f96f6d606ce4e74869e6eef378afb5dc9c0  bitcoin-core-osx-22-res.yml
Done.
```

fanquake commented on src/Makefile.qt.include:
```diff
@@ -360,13 +360,13 @@ translate: $(srcdir)/qt/bitcoinstrings.cpp $(QT_FORMS_UI) $(QT_FORMS_UI) $(BITCO
 	@test -f $(RCC)
 	@cp -f $< $(@D)/temp_$(<F)
-	$(AM_V_GEN) QT_SELECT=$(QT_SELECT) $(RCC) -name bitcoin_locale $(@D)/temp_$(<F) | \
+	$(AM_V_GEN) QT_SELECT=$(QT_SELECT) QT_RCC_SOURCE_DATE_OVERRIDE=1 $(RCC) -name bitcoin_locale --format-version 1 $(@D)/temp_$(<F) | \
```
If we're using `--format-version 1`, can't we drop `QT_RCC_SOURCE_DATE_OVERRIDE` ?

hebasto replied:
Yes! I've overlooked it :man_facepalming:

hebasto replied:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/21654#issuecomment-823347232).

hebasto commented:
Updated 6c05fa9bfcadc7b4be579f6d69c3a758c6153863 -> 712d9b4089eba9b103e67ceed0f65f3861a57cc5 ([pr21654.02](https://github.com/hebasto/bitcoin/commits/pr21654.02) -> [pr21654.03](https://github.com/hebasto/bitcoin/commits/pr21654.03), [diff](https://github.com/hebasto/bitcoin/compare/pr21654.02..pr21654.03)):

- addressed @fanquake's [comment](https://github.com/bitcoin/bitcoin/pull/21654#discussion_r616407949)

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
1a363c5c34b05ed4ca988d7cce105a8f5517df8fcc6b53594d015be518da99b7  guix-build-712d9b4089eb/output/aarch64-linux-gnu/bitcoin-712d9b4089eb-aarch64-linux-gnu-debug.tar.gz
ee34994ec084c2ecbd6026d7ac4d4861f710791b64b7db5f8e524e382bc8070a  guix-build-712d9b4089eb/output/aarch64-linux-gnu/bitcoin-712d9b4089eb-aarch64-linux-gnu.tar.gz
b66f4b90e301cf25ea12ced1fe41b06e25cdd1a96920040a3350e51997ce19e9  guix-build-712d9b4089eb/output/arm-linux-gnueabihf/bitcoin-712d9b4089eb-arm-linux-gnueabihf-debug.tar.gz
09f17f81efdfeab9b05e71276365c9a6f6cba5d4ed9ef98486bd6256ab6a3bc0  guix-build-712d9b4089eb/output/arm-linux-gnueabihf/bitcoin-712d9b4089eb-arm-linux-gnueabihf.tar.gz
df2359e27323f634b0094282d70fef24020048fdaefe673af9037b565dd5d7d7  guix-build-712d9b4089eb/output/dist-archive/bitcoin-712d9b4089eb.tar.gz
7ee887a79509d3f433c96bd4eeb2aea04b7e89a8b77a579fc70ea973c2608ca0  guix-build-712d9b4089eb/output/powerpc64-linux-gnu/bitcoin-712d9b4089eb-powerpc64-linux-gnu-debug.tar.gz
67219c0e8e36ede726e3f4127450c8a13e21b4b775d02761e736d5ec28abc005  guix-build-712d9b4089eb/output/powerpc64-linux-gnu/bitcoin-712d9b4089eb-powerpc64-linux-gnu.tar.gz
baa5cf70e78a00a48d44a0df6a8a1154f00ee03035718f8d51eee6a884ad1f9d  guix-build-712d9b4089eb/output/powerpc64le-linux-gnu/bitcoin-712d9b4089eb-powerpc64le-linux-gnu-debug.tar.gz
357a4e9e6dfd97bc6ec59fa52407f913b23c9afe23b5cd5189f488b2d1bb528a  guix-build-712d9b4089eb/output/powerpc64le-linux-gnu/bitcoin-712d9b4089eb-powerpc64le-linux-gnu.tar.gz
eeac00873e9e7b6bf91a30dba5545e816df85bcc1d5c6ae293104eb7b4750c7c  guix-build-712d9b4089eb/output/riscv64-linux-gnu/bitcoin-712d9b4089eb-riscv64-linux-gnu-debug.tar.gz
c2e1a03a9f9682c124d81e988d07f539bf750deed2f075897e981c81b804e57a  guix-build-712d9b4089eb/output/riscv64-linux-gnu/bitcoin-712d9b4089eb-riscv64-linux-gnu.tar.gz
a6c87f411b047bd95bb66e6269695f3bc352d96dbded3acc06afa5a1ff4c8308  guix-build-712d9b4089eb/output/x86_64-apple-darwin18/bitcoin-712d9b4089eb-osx-unsigned.dmg
9a5888ea9a760f5c8ed4e04671d02b6ae09cdcaac66d9d0b6bcc699dd99465f6  guix-build-712d9b4089eb/output/x86_64-apple-darwin18/bitcoin-712d9b4089eb-osx-unsigned.tar.gz
4bf937edf94c33b9120991b458793c86a2a89954e4f0f04cecc3c179a2fcbb52  guix-build-712d9b4089eb/output/x86_64-apple-darwin18/bitcoin-712d9b4089eb-osx64.tar.gz
12cf391934803a9dc41b0767a1e7d30552824437f003b4d9d902791f3c110158  guix-build-712d9b4089eb/output/x86_64-linux-gnu/bitcoin-712d9b4089eb-x86_64-linux-gnu-debug.tar.gz
81760e22ebffef70c0ba7e3cd28a68a6d6352dff17c261a783f80ac993acf826  guix-build-712d9b4089eb/output/x86_64-linux-gnu/bitcoin-712d9b4089eb-x86_64-linux-gnu.tar.gz
f3d88ff11254d13f5f8cfa65d834e1cfed30d9db2313084ee3d55a09fd1cd033  guix-build-712d9b4089eb/output/x86_64-w64-mingw32/bitcoin-712d9b4089eb-win-unsigned.tar.gz
28a679436baf6faf7fbf165e3ad9504548487713813974f1a80b607eafb422e3  guix-build-712d9b4089eb/output/x86_64-w64-mingw32/bitcoin-712d9b4089eb-win64-debug.zip
91ebf80a73a8444dd1643544ca45f5bebf587220e8666ebe33bed7530c5a8892  guix-build-712d9b4089eb/output/x86_64-w64-mingw32/bitcoin-712d9b4089eb-win64-setup-unsigned.exe
2d2edecde7f6ce65c4db58dcd83ca9eda135fee4cc335c58d409717f0bd61815  guix-build-712d9b4089eb/output/x86_64-w64-mingw32/bitcoin-712d9b4089eb-win64.zip
```
#### Gitian builds:
- Linux
```
Generating report
7e54d537e53b3b9c4a6c3917040f17bc9a90e0023f73854497082b7bf228aeb8  bitcoin-712d9b4089eb-aarch64-linux-gnu-debug.tar.gz
bd17fa107b32f631fac1b29540b92f12d728b66fd1dbb89b79395132a94bab54  bitcoin-712d9b4089eb-aarch64-linux-gnu.tar.gz
54a38c4280b4f244816db82c036bf888af4b55e65147a5394eb91ed5cc013a46  bitcoin-712d9b4089eb-arm-linux-gnueabihf-debug.tar.gz
7575ccde112d732467fe834d2679f9cd12b46322a13a41ef2f4f41818dad4d44  bitcoin-712d9b4089eb-arm-linux-gnueabihf.tar.gz
65b9f23fd131b96c2313688c0d11a8a556e497dec25c4aafec646a843810a470  bitcoin-712d9b4089eb-powerpc64-linux-gnu-debug.tar.gz
390e8d41e9795c70fe1b1722b80cc0d14f0d5d6bfa70b11ded7c5f7f251e3b16  bitcoin-712d9b4089eb-powerpc64-linux-gnu.tar.gz
f4e6cf67e4170900ca94455f9b71e485d7fd42b12844c732e7d69c909b7da1fe  bitcoin-712d9b4089eb-powerpc64le-linux-gnu-debug.tar.gz
630b3da12cb5d7c76047e89cfec3fa5130ffa8cdbbdf38a84f6a3b2e7fe1f3ea  bitcoin-712d9b4089eb-powerpc64le-linux-gnu.tar.gz
643fb8c5e17f00dafeee97bbe590814dfc4476b68ff4c4453cfa45a6d5e7d534  bitcoin-712d9b4089eb-riscv64-linux-gnu-debug.tar.gz
5791c1e179f75b017422ff45cff2552f806df60218c5019ad1ac62d9c11a4e7f  bitcoin-712d9b4089eb-riscv64-linux-gnu.tar.gz
4b4d6efef8766f5aebd779941dd5cd663cad04b282217e14bdbb5e463ee086c5  bitcoin-712d9b4089eb-x86_64-linux-gnu-debug.tar.gz
4a8338c3ef75b3202dff707a24826a34634c834bc8e8818d983985d11e492ebb  bitcoin-712d9b4089eb-x86_64-linux-gnu.tar.gz
df2359e27323f634b0094282d70fef24020048fdaefe673af9037b565dd5d7d7  src/bitcoin-712d9b4089eb.tar.gz
5e9dbd90a9e31deb140557f989965b2ba32ff3c8a723a013515b5aef6a841371  bitcoin-core-linux-22-res.yml
Done.
```
- Windows
```
Generating report
9e10737649eb9d46a8234a8ad2b6d9c27591b40e072801a36ae5be3f39c64ca0  bitcoin-712d9b4089eb-win-unsigned.tar.gz
54877e69a8fe3295e95c641aa907b2d8efe10bcc6c7575775181c8fef484d3cb  bitcoin-712d9b4089eb-win64-debug.zip
cc57404b616a7924bcb905452b87acd1f40968bf592d11c7f04f370e9be9d5aa  bitcoin-712d9b4089eb-win64-setup-unsigned.exe
fe4d1f2015743ac1ed886e486b3836911ce775a0f0552cacf2b2fd0998f1acb8  bitcoin-712d9b4089eb-win64.zip
df2359e27323f634b0094282d70fef24020048fdaefe673af9037b565dd5d7d7  src/bitcoin-712d9b4089eb.tar.gz
cb1dacee26d25e1d07dc57a96e080e0fba2b888cb4956bf19f20b11afcb469ed  bitcoin-core-win-22-res.yml
Done.
```
- macOS
```
Generating report
9412afce70aa79fe3689bbdf50db8fe3d770dbad4a83bc00e1af6b3ed82c924e  bitcoin-712d9b4089eb-osx-unsigned.dmg
4e403edb1df90ec23afdfb792314e78ee941a2fe0a4f304245c738f5960cd3c6  bitcoin-712d9b4089eb-osx-unsigned.tar.gz
a463dedbd514bb741398ce75b7e46bdaedb55afa85df68e34dfb6d473135b0f0  bitcoin-712d9b4089eb-osx64.tar.gz
df2359e27323f634b0094282d70fef24020048fdaefe673af9037b565dd5d7d7  src/bitcoin-712d9b4089eb.tar.gz
f6a4bb0c69bbc3266d0ec8e5642f3c0a3188300db48d5e2f45af4a78ac310d5c  bitcoin-core-osx-22-res.yml
Done.
```

hebasto commented:
Rebased 712d9b4089eba9b103e67ceed0f65f3861a57cc5 -> f17d3406e48ba4ae6e51d2a947636cbf030bc0eb ([pr21654.03](https://github.com/hebasto/bitcoin/commits/pr21654.03) -> [pr21654.04](https://github.com/hebasto/bitcoin/commits/pr21654.04)) due to the conflict with #21655.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
6165661159f0d383eb87004ecc4fa75bdb4ae3fb2ee3a3acfb1435e6ce816e49  guix-build-f17d3406e48b/output/aarch64-linux-gnu/bitcoin-f17d3406e48b-aarch64-linux-gnu-debug.tar.gz
f50f91d8219c6fbe95ff106c90ead63ad724e36fb87b9943770914540dde0cf1  guix-build-f17d3406e48b/output/aarch64-linux-gnu/bitcoin-f17d3406e48b-aarch64-linux-gnu.tar.gz
161f028a9da1547b06ae6eaf5f517c66cce43fa6b41d53612d3ff6da5143d192  guix-build-f17d3406e48b/output/arm-linux-gnueabihf/bitcoin-f17d3406e48b-arm-linux-gnueabihf-debug.tar.gz
8b1a5f9b90862e24e72b95bd4d136bee4c09024e0f7a9bc99cda76a9345ca4fd  guix-build-f17d3406e48b/output/arm-linux-gnueabihf/bitcoin-f17d3406e48b-arm-linux-gnueabihf.tar.gz
0a60041c79a3bf98e4691ad03b37c4a3e29198e4f8ef299735b76f8743b7f9a0  guix-build-f17d3406e48b/output/dist-archive/bitcoin-f17d3406e48b.tar.gz
b1ac9ae12395f21f800d5fccf8b8bc996742ff3c243aade32ee1edf083f4c99b  guix-build-f17d3406e48b/output/powerpc64-linux-gnu/bitcoin-f17d3406e48b-powerpc64-linux-gnu-debug.tar.gz
e2be8b629b6a0c00b8ff31729991d5ceb77b682e3379b1718bf2f6776ba07661  guix-build-f17d3406e48b/output/powerpc64-linux-gnu/bitcoin-f17d3406e48b-powerpc64-linux-gnu.tar.gz
30bf7987a19a50f63d32eb39b0610b71ce56ed6edef15a227a36c7bfdc1b62fb  guix-build-f17d3406e48b/output/powerpc64le-linux-gnu/bitcoin-f17d3406e48b-powerpc64le-linux-gnu-debug.tar.gz
8291496f81da6b843518af14bd91e43978f041964ef51b38369faec1e83e4204  guix-build-f17d3406e48b/output/powerpc64le-linux-gnu/bitcoin-f17d3406e48b-powerpc64le-linux-gnu.tar.gz
80c2f28c5b5326c7a9a773e2da1b28fd7d884a1897a87e0a4e385831e12346c6  guix-build-f17d3406e48b/output/riscv64-linux-gnu/bitcoin-f17d3406e48b-riscv64-linux-gnu-debug.tar.gz
cc03d16bbfedc5a2bb022507eb646d2a11f4dd294641335214bdcc1b4871ec60  guix-build-f17d3406e48b/output/riscv64-linux-gnu/bitcoin-f17d3406e48b-riscv64-linux-gnu.tar.gz
51042f29d28e845e2fa6ef58ebdc6f1d67ad47963aa7b49d134bac8d933d47df  guix-build-f17d3406e48b/output/x86_64-apple-darwin18/bitcoin-f17d3406e48b-osx-unsigned.dmg
c6465fb6791e477835d3b0c13acae504ca6d35ccc2f9d684beccf0a363c0d3da  guix-build-f17d3406e48b/output/x86_64-apple-darwin18/bitcoin-f17d3406e48b-osx-unsigned.tar.gz
bae7103c0ef587515efebaf5aebe2bc5d01bd72ac1f828da26591b04ff60e27c  guix-build-f17d3406e48b/output/x86_64-apple-darwin18/bitcoin-f17d3406e48b-osx64.tar.gz
d912b2811eacb7f91dcb3a92e172f358bf6e51dce432016d5f481f156bca7a06  guix-build-f17d3406e48b/output/x86_64-linux-gnu/bitcoin-f17d3406e48b-x86_64-linux-gnu-debug.tar.gz
d03a3314cf137f2ef68e3e7cfcaeb31ccb5e43ba4d243d6a61fc59b09a89b499  guix-build-f17d3406e48b/output/x86_64-linux-gnu/bitcoin-f17d3406e48b-x86_64-linux-gnu.tar.gz
7b68647e6d935e456e71c29afdb8d58c432ea04ebbd5b68e47bbc0743531690f  guix-build-f17d3406e48b/output/x86_64-w64-mingw32/bitcoin-f17d3406e48b-win-unsigned.tar.gz
fea56fceb0e3b674d1022596af58cbd7e2069bb5b3f615af9f87ed3fbbdf11cb  guix-build-f17d3406e48b/output/x86_64-w64-mingw32/bitcoin-f17d3406e48b-win64-debug.zip
fcd12bbf1e8cfd05d5cb67ee99665d2fe215f94300bf7534085f9384c3f13a19  guix-build-f17d3406e48b/output/x86_64-w64-mingw32/bitcoin-f17d3406e48b-win64-setup-unsigned.exe
ea1b7aa06189d141c20bbf2d5f53b57145ed594802fea54cc45dcedde72df066  guix-build-f17d3406e48b/output/x86_64-w64-mingw32/bitcoin-f17d3406e48b-win64.zip
```
#### Gitian builds:
- Linux
```
Generating report
e0c132e90903fb64aa2603a7752cad535610f430ab97ae37d24677244ed80032  bitcoin-f17d3406e48b-aarch64-linux-gnu-debug.tar.gz
86902a06b4741273d1ebe7c53c245d5ad846c50b88c6def75e182914dde8b9b6  bitcoin-f17d3406e48b-aarch64-linux-gnu.tar.gz
6ab754656622d594c98cffac1c46bc833384e59d6bc6c01962501134f320e23a  bitcoin-f17d3406e48b-arm-linux-gnueabihf-debug.tar.gz
b3ebc13bf827e5b71619e9150461f14d933e2db7eae623b2fa59fee441c5fad7  bitcoin-f17d3406e48b-arm-linux-gnueabihf.tar.gz
3362451f554695be462520d8d9debb237806c71f9085df7aa105369f4412c32e  bitcoin-f17d3406e48b-powerpc64-linux-gnu-debug.tar.gz
3af4fbfd37a3f871443b0d16776e0d480d9e273f2f077cf906c56b0f5286b0a9  bitcoin-f17d3406e48b-powerpc64-linux-gnu.tar.gz
c25f0c62002175152cc0076745e37e97308ea7a49842ba15dae8b53cf42591e6  bitcoin-f17d3406e48b-powerpc64le-linux-gnu-debug.tar.gz
f32517478af6750dad1c0e23a18c1917523397dedfa7e06b5141df3e96c92cde  bitcoin-f17d3406e48b-powerpc64le-linux-gnu.tar.gz
2293a2a3f26cf575e5b8d85670d352524dfb474504a10b165fdd9885abfc0289  bitcoin-f17d3406e48b-riscv64-linux-gnu-debug.tar.gz
84c2cba360f725f1ec136e2cebee828cdd7d6ec75a7d9dc7fd08074adcaac114  bitcoin-f17d3406e48b-riscv64-linux-gnu.tar.gz
ab349482928bc4fa8698c2afff55b8f40bcea0cfdeb87cfeaa9db00648acac1e  bitcoin-f17d3406e48b-x86_64-linux-gnu-debug.tar.gz
d1702d8a4e0413e0dc76161003ea1074afcd5bdf38e11b5a6ea9da5d90e79e75  bitcoin-f17d3406e48b-x86_64-linux-gnu.tar.gz
0a60041c79a3bf98e4691ad03b37c4a3e29198e4f8ef299735b76f8743b7f9a0  src/bitcoin-f17d3406e48b.tar.gz
40d57560dad7424e45b526a726633a16707f0e640144e89f180291506c385237  bitcoin-core-linux-22-res.yml
Done.
```
- Windows
```
Generating report
f74f6787b153358711eeaa6b9b1cd303cf61cc4bea640c50dcf54cc95c5dcc3e  bitcoin-f17d3406e48b-win-unsigned.tar.gz
2ce7e8ba8519d88976e8bba5f889f2341178737272414e5929993df9bf60bfa1  bitcoin-f17d3406e48b-win64-debug.zip
961d95586f911f6f5eef9219a5a4c9e40c25b066a5a3cfc7f8f89b2ae022bf8f  bitcoin-f17d3406e48b-win64-setup-unsigned.exe
abeb15ca648eccf6429580989d1d7a6c5d5ac35f22ca8ea13557787fa878d61a  bitcoin-f17d3406e48b-win64.zip
0a60041c79a3bf98e4691ad03b37c4a3e29198e4f8ef299735b76f8743b7f9a0  src/bitcoin-f17d3406e48b.tar.gz
b9aedd80085aad25f7bc754f1a5c266e2dcd6b5631648161dc8e9c5146f898f5  bitcoin-core-win-22-res.yml
Done.
```
- macOS
```
Generating report
15da5ef2ee1343c07d26d649b31018b2e0f4475126b7ea5ca5cfbd5e0d922883  bitcoin-f17d3406e48b-osx-unsigned.dmg
4f722f2d9c629ad3ec3a62ccf41c3b5a4bf9c6ca3fa3bf0ad266df445da6d21c  bitcoin-f17d3406e48b-osx-unsigned.tar.gz
486113a97a0b71aa7bf18988c326443f91d0c59d45fb0f0a8bbe1943147d0ed0  bitcoin-f17d3406e48b-osx64.tar.gz
0a60041c79a3bf98e4691ad03b37c4a3e29198e4f8ef299735b76f8743b7f9a0  src/bitcoin-f17d3406e48b.tar.gz
6afd06adbede2461fe1eaf616e7f32c2d24a69d6642d4fcd079bf6f0af3afaa4  bitcoin-core-osx-22-res.yml
Done.
```

hebasto commented:
Rebased f17d3406e48ba4ae6e51d2a947636cbf030bc0eb -> a58868d201cb6d263aa552815f7f86562c1ca9a5 ([pr21654.04](https://github.com/hebasto/bitcoin/commits/pr21654.04) -> [pr21654.05](https://github.com/hebasto/bitcoin/commits/pr21654.05)) due to the conflict with #21708.

hebasto commented:
#### Guix builds:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
14f237092a42969b6ab7f1e09393832c5a5ef09d9367d846e73fef1de9cf6d66  guix-build-a58868d201cb/output/aarch64-linux-gnu/bitcoin-a58868d201cb-aarch64-linux-gnu-debug.tar.gz
f6a075d4016c602706bdc78bae94a5951f0213aea790a576841c858d2fe921a7  guix-build-a58868d201cb/output/aarch64-linux-gnu/bitcoin-a58868d201cb-aarch64-linux-gnu.tar.gz
5974cecf082b2a9b4dde2750702de4c5f5dc5bb9309d0048b1fe71e4ba33a517  guix-build-a58868d201cb/output/arm-linux-gnueabihf/bitcoin-a58868d201cb-arm-linux-gnueabihf-debug.tar.gz
48bde6ad02b93e274a8880c33b1f4cb8bb6df44c463aa53b77a6bf7bdf4206e3  guix-build-a58868d201cb/output/arm-linux-gnueabihf/bitcoin-a58868d201cb-arm-linux-gnueabihf.tar.gz
df91fe44f0ff91dbee6a37d4a6ed5d16a7aa91897d4a5d53f5f127f3f29f66ce  guix-build-a58868d201cb/output/dist-archive/bitcoin-a58868d201cb.tar.gz
ccc20363bae8f0b9a99c99b7c544e35cf5f6d3d8682d698e0d6625702271f7e6  guix-build-a58868d201cb/output/powerpc64-linux-gnu/bitcoin-a58868d201cb-powerpc64-linux-gnu-debug.tar.gz
3652776754179279ddceafbd87c2ef5e1eaa50ebe44fdeff2bf2244c9cb92352  guix-build-a58868d201cb/output/powerpc64-linux-gnu/bitcoin-a58868d201cb-powerpc64-linux-gnu.tar.gz
6da405b2004a15120874d8695b5fa0507e851767e533eea0a0f2a9ec7c615e2d  guix-build-a58868d201cb/output/powerpc64le-linux-gnu/bitcoin-a58868d201cb-powerpc64le-linux-gnu-debug.tar.gz
c6403b98cffbba578055625b0ca74a2557ea9f8eb98e370a6ca951a022bac244  guix-build-a58868d201cb/output/powerpc64le-linux-gnu/bitcoin-a58868d201cb-powerpc64le-linux-gnu.tar.gz
ff64f5a94bd36dec9af3b98167f45459bc9a685a3b1c13bed28dc168332e3474  guix-build-a58868d201cb/output/riscv64-linux-gnu/bitcoin-a58868d201cb-riscv64-linux-gnu-debug.tar.gz
032e06edc67248b57480f2194e8f0948017dfcb5dd2460c95ed606191276c23e  guix-build-a58868d201cb/output/riscv64-linux-gnu/bitcoin-a58868d201cb-riscv64-linux-gnu.tar.gz
c1d990e9f84f5dd3e6b46b29f61868de94e7a7f15e9fe1a07d0dffa6c19387e1  guix-build-a58868d201cb/output/x86_64-apple-darwin18/bitcoin-a58868d201cb-osx-unsigned.dmg
cbdf838e18734f596520cd8b066ea875ab3cde667d1ddcbbfb1037dee131d84e  guix-build-a58868d201cb/output/x86_64-apple-darwin18/bitcoin-a58868d201cb-osx-unsigned.tar.gz
8e25aa60395e2a9a45308097b4e5c5e90cf8d2b3c7e603318eb41e5e2a076d2d  guix-build-a58868d201cb/output/x86_64-apple-darwin18/bitcoin-a58868d201cb-osx64.tar.gz
f9d2f04ea142eb805f795a1f378351dbb36153e3e7d9471f8d5ca8d472b4a4a8  guix-build-a58868d201cb/output/x86_64-linux-gnu/bitcoin-a58868d201cb-x86_64-linux-gnu-debug.tar.gz
211c3a7f1a6d7dacb24a414ca50620c59ece8d4df9ec5f7cef67db9279f66060  guix-build-a58868d201cb/output/x86_64-linux-gnu/bitcoin-a58868d201cb-x86_64-linux-gnu.tar.gz
2e5b243c5d54cec8bdba5fb44226344cb2a77a73c125d7c17565f241b3105d7d  guix-build-a58868d201cb/output/x86_64-w64-mingw32/bitcoin-a58868d201cb-win-unsigned.tar.gz
58cc7fb4ac55ed3fd8b33dd8140c8428ef2e5777183ec23d4e78fa662c5de116  guix-build-a58868d201cb/output/x86_64-w64-mingw32/bitcoin-a58868d201cb-win64-debug.zip
88043393d9caa059ab8baf76ea710ca0015c535f6e0a2ed9073b105f00fb30f0  guix-build-a58868d201cb/output/x86_64-w64-mingw32/bitcoin-a58868d201cb-win64-setup-unsigned.exe
0338280ceccb20c7f1bd5858f187eafcd561e0f23d39fad298908d6edb311dde  guix-build-a58868d201cb/output/x86_64-w64-mingw32/bitcoin-a58868d201cb-win64.zip
```
#### Gitian builds:
- Linux
```
Generating report
2b6459626b832e797cdce0118c955af64adf0322c0737b92e6f62537b243a596  bitcoin-a58868d201cb-aarch64-linux-gnu-debug.tar.gz
e44bce285597054fc38e23a787eddda7fcd71107dd3daaa6c9a63138ea38b574  bitcoin-a58868d201cb-aarch64-linux-gnu.tar.gz
5b5f17ea55cf63b35668ed9acbb6542b06b2226c7a3fef92253c71901349d73a  bitcoin-a58868d201cb-arm-linux-gnueabihf-debug.tar.gz
15b78f5d7de418f300050fde01dd78fded6e0fd950658cc3b7a53ed88c201012  bitcoin-a58868d201cb-arm-linux-gnueabihf.tar.gz
018a3de15d708e9afb570bd2445178c05e46165029336e1ad526f5e10658890c  bitcoin-a58868d201cb-powerpc64-linux-gnu-debug.tar.gz
031e9f73e8ba816de24ce6de4588b59b9f3916c752f7baa5c3402eddda9795f9  bitcoin-a58868d201cb-powerpc64-linux-gnu.tar.gz
440f28e8b4311050deef69ef8531038a0413fdb09c42be706a3814345d247bc3  bitcoin-a58868d201cb-powerpc64le-linux-gnu-debug.tar.gz
e32b20544cb96ad9b8703695afb6abf9101cd94dbb4cf46849480c653025b49d  bitcoin-a58868d201cb-powerpc64le-linux-gnu.tar.gz
928ab8fe754fc1df544594609ccd4aca147b53291ec7568d97bb1d752034e102  bitcoin-a58868d201cb-riscv64-linux-gnu-debug.tar.gz
8eb54c320043eb3ff6cabee44bf137c0e9198e8ad1f971116fbb7551d1deea91  bitcoin-a58868d201cb-riscv64-linux-gnu.tar.gz
66f2c49ca08756d2879c7249987e8bbe2085753a058283b06548cc17a68b16a8  bitcoin-a58868d201cb-x86_64-linux-gnu-debug.tar.gz
3bffd8e3f19efeef006e5b55693fa55051b775d2eb64032c0a27e41a52c3e98b  bitcoin-a58868d201cb-x86_64-linux-gnu.tar.gz
df91fe44f0ff91dbee6a37d4a6ed5d16a7aa91897d4a5d53f5f127f3f29f66ce  src/bitcoin-a58868d201cb.tar.gz
9fd4cc1dd653beb8a7d02ee7ed93be4b97bc87955604a7c185d303a981d91df2  bitcoin-core-linux-22-res.yml
Done.
```
- Windows
```
Generating report
98eed8b9abbe524e9e10a512bfb9c1adc808afdea95a622d2f91a4d222f85a53  bitcoin-a58868d201cb-win-unsigned.tar.gz
16043ed7ece90a8cdfcf02e789eedbb5bb675d3fd6d33a30e9dc27603c5f7b43  bitcoin-a58868d201cb-win64-debug.zip
322ca6fbd83f842eb8412e564230eb1d3e7fa1e361b5025db16e7d290cb59bd8  bitcoin-a58868d201cb-win64-setup-unsigned.exe
535a53a1a0e0a348bb2847d22c7b8e9640896a959ac7b1e2d98ce16271eaccab  bitcoin-a58868d201cb-win64.zip
df91fe44f0ff91dbee6a37d4a6ed5d16a7aa91897d4a5d53f5f127f3f29f66ce  src/bitcoin-a58868d201cb.tar.gz
42432b649ff1c63b89b3c6dc2199fdd0918c43bdc226b53377e96d54624fd144  bitcoin-core-win-22-res.yml
Done.
```
- macOS
```
Generating report
4584f612566684a7fabb8ccab77d287638e079566dd0ff88d0b3ce111c737e36  bitcoin-a58868d201cb-osx-unsigned.dmg
172b8e4ba6f005dc0980a7ec8ef7391931ca3483693d3bfd9442e36d61c1b396  bitcoin-a58868d201cb-osx-unsigned.tar.gz
664e3577c09dcddcff314c7fc018c46986b75f835cc360d62db3671bea706bd4  bitcoin-a58868d201cb-osx64.tar.gz
df91fe44f0ff91dbee6a37d4a6ed5d16a7aa91897d4a5d53f5f127f3f29f66ce  src/bitcoin-a58868d201cb.tar.gz
deebfab06bf6a99a69f4a9efdea89515823f161ffba485e884fb5080d4052485  bitcoin-core-osx-22-res.yml
Done.
```

fanquake commented:
@laanwj do you have any opinion on passing `--format-version 1` to RCC?

laanwj commented:
> @laanwj do you have any opinion on passing --format-version 1 to RCC?

Not really. If it improves determinism that's a good thing. I think the only thing to be careful of is that if a new version of Qt adds a new format that is better somehow and differs in more than just this comment, we need to remember to reevaluate this.

hebasto commented:
@laanwj 
> > @laanwj do you have any opinion on passing --format-version 1 to RCC?
> 
> Not really. If it improves determinism that's a good thing. I think the only thing to be careful of is that if a new version of Qt adds a new format that is better somehow and differs in more than just this comment, we need to remember to reevaluate this.

Do you think the usage of `QT_RCC_SOURCE_DATE_OVERRIDE=1` is safer? (see https://github.com/bitcoin/bitcoin/pull/21654#issuecomment-819198228)

laanwj commented:
> Do you think the usage of QT_RCC_SOURCE_DATE_OVERRIDE=1 is safer? (see #21654 (comment))

I don't have a strong opinion on it. I think it's fine to change this.

fanquake reviewed (approved):
ACK a58868d201cb6d263aa552815f7f86562c1ca9a5
